app/src/main/java/com/example/habitsmasher/ui/dashboard/EditHabitFragment.java

```diff
@@ -0,0 +1,131 @@
+package com.example.habitsmasher.ui.dashboard;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.DialogFragment;
+
+import com.example.habitsmasher.Habit;
+import com.example.habitsmasher.R;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Collection;
+import java.util.Date;
+import java.util.HashMap;
+
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+public class EditHabitFragment extends DialogFragment {
+    private EditText _titleText;
+    private EditText _reasonText;
+    private EditText _dateText;
+    //private TextView _errorText;
+    private TextView _header;
+
+    private Button _okButton;
```

Comment:
To avoid confusion, this button is actually **confirm** on the front-end, so we can rename this field to `_confirmButton`